src/lib.rs

```diff
@@ -95,8 +95,9 @@
 
 #![forbid(rust_2018_idioms)]
 #![deny(missing_debug_implementations, nonstandard_style)]
-#![warn(missing_docs, missing_doc_code_examples, unreachable_pub)]
```

Comment:
Why remove this? better would add an example for it.

Reply:
That'd be several hundred lines, which is out of scope for this PR.